src/clients/api/mutations/vote/useCastVote.ts

```diff
@@ -18,7 +19,28 @@ const useCastVote = (
         fromAccountAddress,
         ...params,
       }),
-    options,
+    {
+      ...options,
+      onSuccess: (...onSuccessParams) => {
+        const { proposalId, voteType } = onSuccessParams[1];
+
+        // Invalidate query to fetch voters
+        queryClient.invalidateQueries([
+          FunctionKey.GET_VOTERS,
+          {
+            id: proposalId,
+            filter: voteType,
+          },
+        ]);
+
+        // Invalidate query to fetch proposal list
+        queryClient.invalidateQueries([FunctionKey.GET_PROPOSALS]);
```

Comment:
This is not necessary because this query polls while the proposal is active.

Reply:
Although we automatically refresh the query on an interval, it is unlikely to be synced exactly with when a new block is mined (since the polling interval is just set as soon as the query mounts), meaning users could have to wait up to a full interval (15 seconds currently) before seeing the UI update